engine/src/main/java/io/camunda/zeebe/engine/state/appliers/ProcessInstanceCreationCreatedApplier.java

```diff
@@ -0,0 +1,107 @@
+    this.elementInstanceState = elementInstanceState;
+  }
+
+  @Override
+  public void applyState(final long key, final ProcessInstanceCreationRecord value) {
+    if (value.hasStartInstructions()) {
+      final var process =
+          processState.getProcessByKey(value.getProcessDefinitionKey()).getProcess();
+      final ElementInstance processInstance =
+          elementInstanceState.getInstance(value.getProcessInstanceKey());
+
+      value.getStartInstructions().stream()
+          .map(instruction -> process.getElementById(instruction.getElementId()))
+          .filter(element -> element.getElementType().equals(BpmnElementType.PARALLEL_GATEWAY))
+          .map(ExecutableFlowNode.class::cast)
+          .forEach(
+              element -> {
+                final var parentElementId = element.getFlowScope().getId();
+                final ElementInstance flowScope =
+                    findParentFlowScope(processInstance, parentElementId);
+                incrementNumberOfTakenSequenceFlows(element, flowScope);
+              });
+    }
+  }
+
+  /**
+   * Traverses the element instances to find one that matches the parent element id. If this is the
+   * process instance, it is returned immediately. This will work because we will always activate
+   * the flow scope of a start instruction once.
+   *
+   * @param processInstance the highest element instance of a process
+   * @param targetElementId the id we are looking for
+   * @return the element instance which matches the targetElementId
+   */
+  private ElementInstance findParentFlowScope(
+      final ElementInstance processInstance, final DirectBuffer targetElementId) {
+    final ElementInstance flowScope;
+    if (processInstance.getValue().getElementIdBuffer().equals(targetElementId)) {
+      flowScope = processInstance;
+    } else {
+      flowScope = findFlowScopeInChildren(processInstance, targetElementId);
+    }
+    return flowScope;
+  }
+
+  private ElementInstance findFlowScopeInChildren(
+      final ElementInstance processInstance, final DirectBuffer targetElementId) {
+    ElementInstance found = null;
+    final List<ElementInstance> children =
+        elementInstanceState.getChildren(processInstance.getKey());
+
+    for (final ElementInstance childInstance : children) {
+      if (childInstance.getValue().getElementIdBuffer().equals(targetElementId)) {
+        found = childInstance;
+        break;
+      } else {
+        found = findFlowScopeInChildren(childInstance, targetElementId);
+        if (found != null) {
+          break;
+        }
+      }
+    }
+
+    return found;
+  }
+
+  private void incrementNumberOfTakenSequenceFlows(
+      final ExecutableFlowNode element, final ElementInstance flowScope) {
+    element
+        .getIncoming()
+        .forEach(
+            incoming ->
+                elementInstanceState.incrementNumberOfTakenSequenceFlows(
+                    flowScope.getKey(), element.getId(), incoming.getId()));
```

Comment:
🔧 Please update the PR description, to make it clear that we now have to do this as well.